File: app/src/main/java/org/pocketx/knell/base/BaseActivity.java
package org.pocketx.knell.base;

import android.support.v7.app.AppCompatActivity;

import org.pocketx.knell.domain.BirthdayManager;
import org.pocketx.knell.utils.Injector;

/**
 * @author Shadow
 * @author shenghaiyang
 * @date 2017/10/22 14:16
 */

public abstract class BaseActivity extends AppCompatActivity {

    protected final BirthdayManager getBirthdayManager() {
        return Injector.obtain(getApplication());
    }

}

File: app/src/main/java/org/pocketx/knell/base/BaseActivity.kt
package org.pocketx.knell.base

import androidx.appcompat.app.AppCompatActivity

import org.pocketx.knell.domain.BirthdayManager
import org.pocketx.knell.utils.Injector

/**
 * Activity基类
 *
 * @author Shadow
 * @author shenghaiyang
 */
abstract class BaseActivity : AppCompatActivity() {

    protected val birthdayManager: BirthdayManager
        get() = Injector.obtain(application)

}
